flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/OperatorCoordinatorCheckpoints.java

```diff
@@ -121,14 +123,17 @@ private static void acknowledgeAllCoordinators(
                 final Throwable error =
                         checkpoint.isDisposed() ? checkpoint.getFailureCause() : null;
 
+                CheckpointFailureReason reason = CheckpointFailureReason.TRIGGER_CHECKPOINT_FAILURE;
```

Comment:
I do not really follow the idea of this method(the original one). As I understand method `acknowledgeAllCoordinators` can be called only after all taskManagers finish the checkpoint it means that we can check `checkpoint.getFailureCause()` before the loop and throw the exception if needed. I also don't understand why if we already have the `CheckpointException`(from checkpoint.getFailureCause()) with specific `ReasonType` we always wrap it by one more `CheckpointException` with the constant type `TRIGGER_CHECKPOINT_FAILURE`.
Perhaps, @zlzhang0122 or @pnowojski can explain it?

Reply:
From what I can see `acknowledgeAllCoordinators` is called before (CheckpointCoordinator:L578) triggerCheckpoint RPC is called on Task Managers (CheckpointCoordinator:L623). So this happens very early in the in the checkpoint triggering process and is consistent with out `TRIGGER_CHECKPOINT_FAILURE` usages.

Reply:
I agree with @pnowojski that `acknowledgeAllCoordinators` is called very early, and as far as I can see, `acknowledgeAllCoordinators` is used to take checkpoint for `OperatorCoordinator` which is runing on JobMaster. So maybe we can simply consider it as another kind of masterStates.